src/sidebar.cpp

```diff
@@ -0,0 +1,573 @@
+        morale_str = ":D";
+    } else if( has_trait( trait_THRESH_FELINE ) && morale_cur >= 10 ) {
+        morale_str = ":3";
+    } else if( !has_trait( trait_THRESH_FELINE ) && morale_cur >= 10 ) {
+        morale_str = ":)";
+    } else if( morale_cur >= -10 ) {
+        morale_str = ":|";
+    } else if( morale_cur >= -50 ) {
+        morale_str = "):";
+    } else if( morale_cur >= -100 ) {
+        morale_str = "D:";
+    } else {
+        morale_str = "D8";
+    }
+    mvwprintz( w, sideStyle ? 0 : 3, sideStyle ? 11 : 9, col_morale, morale_str );
+
+    vehicle *veh = g->remoteveh();
+    if( veh == nullptr && in_vehicle ) {
+        veh = g->m.veh_at( pos() );
+    }
+    if( veh ) {
+        veh->print_fuel_indicators( w, sideStyle ? 2 : 3, sideStyle ? getmaxx( w ) - 5 : 49 );
+        nc_color col_indf1 = c_light_gray;
+
+        float strain = veh->strain();
+        nc_color col_vel = strain <= 0 ? c_light_blue :
+                           ( strain <= 0.2 ? c_yellow :
+                             ( strain <= 0.4 ? c_light_red : c_red ) );
+
+        // Draw the speedometer.
+        int speedox = sideStyle ? 0 : 28;
+        int speedoy = sideStyle ? 5 :  3;
+
+        bool metric = get_option<std::string>( "USE_METRIC_SPEEDS" ) == "km/h";
+        // Logic below is not applicable to translated units and should be changed
+        int velx    = metric ? 4 : 3; // strlen(units) + 1
+        int cruisex = metric ? 9 : 8; // strlen(units) + 6
+
+        if( !sideStyle ) {
+            if( !veh->cruise_on ) {
+                speedox += 2;
+            }
+            if( !metric ) {
+                speedox++;
+            }
+        }
+
+        const char *speedo = veh->cruise_on ? "%s....>...." : "%s....";
+        mvwprintz( w, speedoy, speedox,        col_indf1, speedo, velocity_units( VU_VEHICLE ) );
+        mvwprintz( w, speedoy, speedox + velx, col_vel,   "%4d",
+                   int( convert_velocity( veh->velocity, VU_VEHICLE ) ) );
+        if( veh->cruise_on ) {
+            mvwprintz( w, speedoy, speedox + cruisex, c_light_green, "%4d",
+                       int( convert_velocity( veh->cruise_velocity, VU_VEHICLE ) ) );
+        }
+
+        const int vel_offset = 11 + ( veh->velocity != 0 ? 2 : 0 );
+        wmove( w, sideStyle ? 4 : 3, getmaxx( w ) - vel_offset );
+        if( veh->velocity != 0 ) {
+            nc_color col_indc = veh->skidding ? c_red : c_green;
+            int dfm = veh->face.dir() - veh->move.dir();
+
+            if( dfm == 0 ) {
+                wprintz( w, col_indc, "^" );
+            } else if( dfm < 0 ) {
+                wprintz( w, col_indc, "<" );
+            } else {
+                wprintz( w, col_indc, ">" );
+            }
+
+            wprintz( w, c_white, " " );
+        }
+
+        //Vehicle direction indicator in 0-359° where 0 is north (veh->face.dir() 0° is west)
+
+        wprintz( w, c_white, string_format( "%3d°", ( veh->face.dir() + 90 ) % 360 ).c_str() );
```

Comment:
`wprintz` does its own formatting, you don't need to call `string_format` before calling it.

Note that the curses functions also take `std::string` as argument now, so the `.c_str` is not needed either. It's not needed for the `string_format` function.